packages/cli-kit/src/public/node/themes/api.ts

```diff
@@ -28,15 +30,48 @@ export type ThemeParams = Partial<Pick<Theme, 'name' | 'role' | 'processing' | '
 export type AssetParams = Pick<ThemeAsset, 'key'> & Partial<Pick<ThemeAsset, 'value' | 'attachment'>>
 
 export async function fetchTheme(id: number, session: AdminSession): Promise<Theme | undefined> {
-  const response = await request('GET', `/themes/${id}`, session, undefined, {fields: 'id,name,role,processing'})
-  return buildTheme(response.json.theme)
+  const response = await adminRequestDoc(GetTheme, session, {id: composeThemeGid(id)})
+
+  const {theme} = response
+  if (!theme) {
+    return undefined
+  }
+  return buildTheme({
+    id: parseGid(theme.id),
+    processing: theme.processing,
+    role: theme.role.toLowerCase(),
+    name: theme.name,
+  })
 }
 
 export async function fetchThemes(session: AdminSession): Promise<Theme[]> {
-  const response = await request('GET', '/themes', session, undefined, {fields: 'id,name,role,processing'})
-  const themes = response.json?.themes
-  if (themes?.length > 0) return themes.map(buildTheme)
-  return []
+  const themes: Theme[] = []
+  let after: string | null = null
+
+  while (true) {
+    // eslint-disable-next-line no-await-in-loop
+    const response = await adminRequestDoc(GetThemes, session, {after})
+    if (!response.themes) {
+      unexpectedGraphQLError('Failed to fetch themes')
+    }
+    const {nodes, pageInfo} = response.themes
+    nodes.forEach((theme) => {
+      const t = buildTheme({
+        id: parseGid(theme.id),
+        processing: theme.processing,
+        role: theme.role.toLowerCase(),
+        name: theme.name,
+      })
+      if (t !== undefined) {
```

Comment:
Any reason for the `!== undefined` check?

Could we do

```suggestion
      if (t) {
```

or can typescript not infer that theme is `Theme` without this?

Reply:
I'm not sure! Let me see if I can figure out an answer.

Reply:
Yeah, it looks like `if (t)` should work here, since `buildTheme` returns either `Theme` or `undefined`.